Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -95,33 +95,41 @@ static bool validateCreateBuffer(const Device& device, const WGPUBufferDescripto
     return true;
 }
 
-static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, WGPUBufferUsageFlags usage, bool mappedAtCreation)
-{
-    if (deviceHasUnifiedMemory)
-        return MTLStorageModeShared;
-#if PLATFORM(MAC) || PLATFORM(MACCATALYST)
-    if (usage & (WGPUBufferUsage_MapRead | WGPUBufferUsage_MapWrite | WGPUBufferUsage_Index))
-        return MTLStorageModeManaged;
-    if (mappedAtCreation)
-        return MTLStorageModeManaged;
-#else
-    UNUSED_PARAM(mappedAtCreation);
-    UNUSED_PARAM(usage);
-#endif
-    return MTLStorageModePrivate;
-}
-
 id<MTLBuffer> Device::safeCreateBuffer(NSUInteger length, MTLStorageMode storageMode, MTLCPUCacheMode cpuCacheMode, MTLHazardTrackingMode hazardTrackingMode) const
 {
     MTLResourceOptions resourceOptions = (cpuCacheMode << MTLResourceCPUCacheModeShift) | (storageMode << MTLResourceStorageModeShift) | (hazardTrackingMode << MTLResourceHazardTrackingModeShift);
-    id<MTLBuffer> buffer = [m_device newBufferWithLength:std::max<NSUInteger>(1, length) options:resourceOptions];
+
+    length = std::max<NSUInteger>(1, length);
+#define WEBGPU_USE_MTL_HEAP 1
```

Comment:
we could remove the define, but I wanted to leave it to locally compare any performance impact to using MTLHeap